src/graphics/instance.rs

```diff
@@ -222,3 +222,20 @@ impl InstanceArray {
         self.capacity as usize
     }
 }
+
+impl<'a> Drawable for &'a mut InstanceArray {
+    fn draw(self, canvas: &mut Canvas, param: DrawParam) {
+        self.flush_wgpu(&canvas.wgpu);
+        canvas.push_draw(
+            Draw::MeshInstances {
+                mesh: canvas.default_resources().mesh.clone(),
+                instances: (&*self).into(),
+            },
+            param,
+        );
+    }
+
+    fn dimensions(self, _gfx: &mut GraphicsContext) -> Option<Rect> {
+        None
+    }
```

Comment:
How about taking over [the current implementation](https://github.com/ggez/ggez/blob/ab8e9e81ed93dcbec6ba14e3b83d4716e6fdff90/src/graphics/spritebatch.rs#L219-L234)?